crates/db_views/src/post/post_view.rs

```diff
@@ -502,36 +499,27 @@ impl<'a> PostQuery<'a> {
       }
 
       query = query.filter(filter_blocked());
-      if let Some(person_id) = o.local_user.person_id() {
-        let blocked_keywords: Vec<String> = user_post_keyword_block::table
-          .filter(user_post_keyword_block::person_id.eq(person_id))
-          .select(user_post_keyword_block::keyword)
+      if o.local_user.is_some() {
+        let blocked_keywords: Vec<String> = local_user_keyword_block::table
+          .filter(
+            local_user_keyword_block::local_user_id
+              .nullable()
+              .eq(my_local_user_id),
+          )
+          .select(local_user_keyword_block::keyword)
           .load::<String>(conn)
           .await?;
```

Comment:
Use `LocalUserKeywordBlock::read()`. For that you need to move the line `let conn = &mut get_conn(pool).await?;` down to the bottom of this function.